src/main/java/net/dv8tion/jda/internal/JDAImpl.java

```diff
@@ -1020,16 +1017,19 @@ public void resetGatewayUrl()
         this.gatewayUrl = null;
     }
 
-    public ScheduledThreadPoolExecutor getAudioLifeCyclePool()
+    public ScheduledExecutorService getAudioLifeCyclePool()
     {
-        ScheduledThreadPoolExecutor pool = audioLifeCyclePool;
+        ScheduledExecutorService pool = threadConfig.getAudioPool();
         if (pool == null)
         {
             synchronized (audioLifeCycleLock)
             {
-                pool = audioLifeCyclePool;
+                pool = threadConfig.getAudioPool();
                 if (pool == null)
-                    pool = audioLifeCyclePool = ThreadingConfig.newScheduler(1, this::getIdentifierString, "AudioLifeCycle");
+                {
+                    pool = ThreadingConfig.newScheduler(1, this::getIdentifierString, "AudioLifeCycle");
+                    threadConfig.setAudioPool(pool, threadConfig.isShutdownAudioPool());
+                }
```

Comment:
This is not a thread-safe solution

Reply:
I think it would be better to just make this a lazy initialization in `ThreadConfig#getAudioPool`.

Reply:
Good point. I'm just realizing now, that it already initializes other pools as well, so moving the audio pool init there should be fine.